src/mono/wasm/host/BrowserHost.cs

```diff
@@ -68,6 +68,14 @@ private async Task RunAsync(ILoggerFactory loggerFactory, CancellationToken toke
                                                debugging: _args.CommonConfig.Debugging);
         runArgsJson.Save(Path.Combine(_args.CommonConfig.AppPath, "runArgs.json"));
 
+        // Read system environment variables after runArgs.json is saved, so they won't be passed to browser.
```

Comment:
Could we instead explicitly use `Environment.GetEnvironmentVariable("ASPNETCORE_URLS")` as fallback in the line below ?

Reply:
That would mean that the URL can't set in runtimeconfig 🤷‍♂️
In general, we have too many options in this area

Reply:
> That would mean that the URL can't set in runtimeconfig 

Sorry I don't follow, please elaborate

Reply:
There so logic that populates the `envvars` collection here, something comes from runtimeconfig, something from command line args. If we read URLs only from environment.envvars we these sources. Since this has to go .NET 9, I want make the smallest change and we can do some cleanup later

Reply:
```
        string? aspnetUrls = null;
        if(!envVars.TryGetValue("ASPNETCORE_URLS", out string? aspnetUrls))
        {
            aspnetUrls = Environment.GetEnvironmentVariable("ASPNETCORE_URLS");
        }
        string[] urls = (!string.IsNullOrEmpty(aspnetUrls))
                            ? aspnetUrls.Split(';', StringSplitOptions.RemoveEmptyEntries)
                            : new string[] { $"http://127.0.0.1:{_args.CommonConfig.HostProperties.WebServerPort}", "https://127.0.0.1:0" };
```

Reply:
or do you want the overwrite the other way